script.js

```diff
@@ -1,4 +1,149 @@
+var GAME_MODE_DICE_ROLL = "GAME_MODE_DICE_ROLL";
+var GAME_MODE_CHOOSE_DICE_ORDER = "GAME_MODE_CHOOSE_DICE_ORDER";
+
+// Initialise the game to start with the dice roll game mode
+var gameMode = GAME_MODE_DICE_ROLL;
+
+// Keep track of the current player's number, either 1 or 2.
+// The game starts with Player 1.
+var currPlayer = 1;
+
+// Creating array to Keep track of each player's dice rolls
+var player1Dice = [];
+var player2Dice = [];
+
+// Keep track of each player's chosen numbers
+var player1Num;
+var player2Num;
+
+// Base
+// Requirements
+// a. There are 2 players and players take turns.
+//Player 1, Player 2.
+
+// b. When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+//Roll Dice function
+
+// Roll dice:  generate random number from 1 to 6
+var getDiceRoll = function () {
+  return Math.ceil(Math.random() * 6);
+};
+
+//Add new dice roll to current player
+
+var getDiceRolls = function () {
+  // Create an array newDiceRolls with 2 independent dice roll values
+  var newDiceRolls = [getDiceRoll(), getDiceRoll()];
+
+  // Assign newDiceRolls to the current player's dice array
+  if (currPlayer === 1) {
+    player1Dice = newDiceRolls;
+  } else {
+    player2Dice = newDiceRolls;
+  }
+  //return new diceroll to main
+  return newDiceRolls;
+};
+
+/**
+ * Return a number that is the concatenation of num1 and num2
+ */
+var concatenate2Numbers = function (num1, num2) {
+  return Number(String(num1) + String(num2));
+};
+
+/**
+ * Generate and store the player's number based on his dice rolls and chosen first numeral index
+ * Return the player number
+ */
+var getPlayerNumber = function (firstNumeralIndex) {
+  // Current player's dice array.
+  var diceArray;
+  if (currPlayer === 1) {
+    diceArray = player1Dice;
+  } else {
+    diceArray = player2Dice;
+  }
+
+  var playerNum;
+  // If the chosen first numeral index is 1, create player number starting with 1st dice
+  if (firstNumeralIndex === 1) {
+    playerNum = concatenate2Numbers(diceArray[0], diceArray[1]);
+  }
+  // Otherwise, create player number starting with 2nd dice
+  else {
+    playerNum = concatenate2Numbers(diceArray[1], diceArray[0]);
+  }
+
+  // Store player num in the relevant global player num variable
+  if (currPlayer === 1) {
+    player1Num = playerNum;
+  } else {
+    player2Num = playerNum;
+  }
+
+  // Return generated player num to parent function
+  return playerNum;
+};
+//Return the number of the winner: 1 or 2 (bigger wins).
+var determineWinner = function () {
+  if (player1Num > player2Num) {
+    return 1;
+  }
+  return 2;
+};
+
```

Comment:
Good use of helper functions so far. They've taken a significant amount of implementation logic out of main, leaving main much neater! 💪🏻